java/src/edu/wpi/first/wpilibj/networktables/NetworkTable.java

```diff
@@ -864,8 +864,8 @@ else if (value instanceof double[])
       return NetworkTablesJNI.putDoubleArray(pathWithSep + key, (double[])value);
     else if (value instanceof Boolean[])
       return NetworkTablesJNI.putBooleanArray(pathWithSep + key, toNative((Boolean[])value));
-    else if (value instanceof Double[])
-      return NetworkTablesJNI.putDoubleArray(pathWithSep + key, toNative((Double[])value));
+    else if (value instanceof Number[])
+      return NetworkTablesJNI.putDoubleArray(pathWithSep + key, toNative((Number[])value));
```

Comment:
We do exactly that... take a look at the implementation of (and the changes in this PR to) `toNative()`.